Add generalized pip installation method

VersusFacit opened the pull request:
Oh hai community team 👋 

## Description & motivation
Was prompted by a question in [community Slack](https://getdbt.slack.com/archives/CBSQTAPLG/p1644358335472189?thread_ts=1644346925.356309&cid=CBSQTAPLG). A community member inquired how non-PyPi hosted adapters can be installed. We were missing an explicit reference to this on the page (prior to this PR there does exist an implicit reference at the bottom of the community adapter table)

- added `git+` installation method to adapter installation page
- generalized some language to be more inclusive
- added a reference to the #adapter-ecosystem channel for future contributors

## Pre-release docs
Is this change related to an unreleased version of dbt?
- [ ] Yes: please
    - update the base branch to `next`
    - add Changelog components: `<Changelog>[New/Changed] in v0.x.0</Changelog>`
    - add links to the "New and changed documentation" section of the latest [Migration Guide](../website/docs/docs/guides/migration-guide)
- [X] No: please ensure the base branch is `current`
- [ ] Unsure: we'll let you know!

## Checklist
:

VersusFacit committed:
Add generalized pip installation method

- added `git+` installation method
- generalized some language to be more inclusive
- added a reference to the #adapter-ecosystem channel for contributors

runleonarun commented on website/docs/docs/available-adapters.md:
````diff
@@ -5,7 +5,15 @@ id: "available-adapters"
+Some vendor or community adapters may not exist in PyPi. However, an adapter hosted on GitHub can still be installed with `pip`:
+
+```shell
+pip install git+https://github.com/{MAINTAINER_NAME}/{ADAPTER_NAME}.git
````
I think we can just use the ALL CAPS and drop the curly brackets. 
```suggestion
pip install git+https://github.com/MAINTAINER_NAME/ADAPTER_NAME.git
```

runleonarun commented on website/docs/docs/available-adapters.md:
```diff
@@ -5,7 +5,15 @@ id: "available-adapters"
+
+Most adapters can be installed from PyPi using `pip`. The installation will include `dbt-core` and any other required dependencies, which may include other adapter plugins. Read more about [installing dbt](dbt-cli/install/overview).
+
+Some vendor or community adapters may not exist in PyPi. However, an adapter hosted on GitHub can still be installed with `pip`:
```
@VersusFacit what do you think about defining the user input? Feel free to enhance my best guesses here:
```suggestion
Some vendor or community adapters may not exist in PyPi. However, you can still install an adapter hosted on GitHub with `pip install`, by replacing MAINTAINER_NAME with the person or company maintaining the adapter on GitHub and ADAPTER_NAME with the git repository's name (these can be taken directly from the package's url):
```

VersusFacit replied:
I erred on the side of "let the code say it" as a guess of what to do. More than happy to go with an explicit definition of terms.

runleonarun reviewed (changes requested):
@VersusFacit This is great! I suggested a few little tweaks. 

I think it's OK to add this to `current`, but heads up that @amychen1776 and I are working on [this PR](https://github.com/dbt-labs/docs.getdbt.com/pull/1036) to revise the section. I can cherry pick changes over to that PR if you want to merge these changes sooner than later. OR we can change the base to `adapter_content` and merge it over there with that work.

VersusFacit committed:
Update website/docs/docs/available-adapters.md

Co-authored-by: Leona B. Campbell <3880403+runleonarun@users.noreply.github.com>

VersusFacit committed:
Update website/docs/docs/available-adapters.md

Co-authored-by: Leona B. Campbell <3880403+runleonarun@users.noreply.github.com>

VersusFacit commented:
@runleonarun Whatever is easy, truly! I'm happy to do the work of merging this into your working branch if wanted. No need to "rush" this in. :) It was just something a community member asked about and that I realized was sort of secret lore in my head that might as well be documented explicitly!

VersusFacit committed:
Update available-adapters.md

Used the work package and I'm sure that will confused people. Fixing that real quick.

joellabes commented on website/docs/docs/available-adapters.md:
```diff
@@ -50,7 +58,7 @@ These adapter plugins are contributed and maintained by members of the community
 | Athena                 | [Profile Setup](athena-profile)       | Athena engine version 2   | `pip install git+https://github.com/Tomme/dbt-athena.git` |
 
 Community-supported plugins are works in progress, and all users are encouraged to contribute by testing and writing code. If you're interested in contributing:
```
```suggestion
Community-supported plugins are works in progress, and anyone is welcome to contribute by testing and writing code. If you're interested in contributing:
```

VersusFacit committed:
Update website/docs/docs/available-adapters.md

Co-authored-by: Joel Labes <joel.labes@dbtlabs.com>

runleonarun reviewed (approved):
@VersusFacit On thinking about this a bit, let's merge this now and I will bring it over to the adapter content reorg.
